File: 2023-2024/sourcecode/mod3/lab8/LiveDataBasicsJava/app/src/main/java/com/example/livedatabasics_java/viewmodel/CounterViewModel.java
package com.example.livedatabasics_java.viewmodel;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

public class CounterViewModel extends ViewModel {
    private final MutableLiveData<Integer> counter = new MutableLiveData<>(0);

    public LiveData<Integer> getCounter() {
        return counter;
    }

    public void increase(){
        if (counter.getValue() != null)
            counter.setValue(counter.getValue() + 1);
    }

    public void decrease(){
        if (counter.getValue() != null)
            counter.setValue(counter.getValue() - 1);
    }

    public void clear(){
        if (counter.getValue() != null)
            counter.setValue(0);
    }
}

File: 2023-2024/sourcecode/mod2/Lab7/LiveDataBasicsKotlin/app/src/main/java/com/example/livedatabasics_kotlin/viewmodel/CounterViewModel.kt
package com.example.livedatabasics_kotlin.viewmodel

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel

class CounterViewModel : ViewModel() {
    private var _counter = MutableLiveData(0)
    val counter: LiveData<Int>
        get() = _counter

    fun increase(){
        _counter.value = _counter.value?.inc()
    }

    fun decrease(){
        _counter.value = _counter.value?.dec()
    }

    fun clear(){ _counter.value = 0 }
}